test/test.cpp

```diff
@@ -36,7 +36,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 unit_test_t _g_unit_tests[1024];
 int _g_num_unit_tests = 0;
-int _g_test_failures = 0;
+int _g_test_failures = 0; // flushed at start of every unit
+int _g_test_total_failures = 0;
```

Comment:
actually, aren't the number of failures stored per unit test in the ``_g_unit_tests`` array?
I would expect the ``test_total_failures`` state is redundant, and could be recomputed from the test array. I haven't looked into this recently, but I suspect there's a smaller/simpler patch to solve the problem. Btw, what was the symptom you experienced?